gcloud/pubsub/connection.py

```diff
@@ -51,8 +51,11 @@ class Connection(base_connection.JSONConnection):
     def __init__(self, credentials=None, http=None, api_base_url=None):
         super(Connection, self).__init__(credentials=credentials, http=http)
         if api_base_url is None:
-            api_base_url = os.getenv(PUBSUB_EMULATOR,
-                                     self.__class__.API_BASE_URL)
+            emulator_host = os.getenv(PUBSUB_EMULATOR)
+            if emulator_host is None:
+                api_base_url = self.__class__.API_BASE_URL
+            else:
+                api_base_url = 'http://' + emulator_host
```

Comment:
I agree, no need for the defensive check.
On Jan 14, 2016 3:51 PM, "Danny Hermes" <notifications@github.com> wrote:

> In gcloud/pubsub/connection.py
> <https://github.com/GoogleCloudPlatform/gcloud-python/pull/1371#discussion_r49783137>
> :
>
> > @@ -51,8 +51,11 @@ class Connection(base_connection.JSONConnection):
> >      def __init__(self, credentials=None, http=None, api_base_url=None):
> >          super(Connection, self).__init__(credentials=credentials, http=http)
> >          if api_base_url is None:
> > -            api_base_url = os.getenv(PUBSUB_EMULATOR,
> > -                                     self.__class__.API_BASE_URL)
> > +            emulator_host = os.getenv(PUBSUB_EMULATOR)
> > +            if emulator_host is None:
> > +                api_base_url = self.__class__.API_BASE_URL
> > +            else:
> > +                api_base_url = 'http://' + emulator_host
>
> I don't think so. The env. var. will usually come from running
>
> $(gcloud beta emulators pubsub env-init)
>
> so a user doesn't actually type anything.
>
> @hbchai <https://github.com/hbchai> care to weigh in?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/GoogleCloudPlatform/gcloud-python/pull/1371/files#r49783137>
> .
>